common/src/types/token.rs

```diff
@@ -701,4 +704,15 @@ impl Token {
                 )
             })
     }
+
+    /// Gets the amount of the token as an f64, accounting for the associated
+    /// number of decimals.
+    ///
+    /// Note that due to conversion to f64, the result may lose precision.
+    pub fn get_amount_f64(&self, amount: u128) -> Result<f64, String> {
```

Comment:
nit: I'd suggest renaming to `apply_decimal_adjustment`, `convert_decimals`, or similar. `get_amount` seems to imply a getter of a property of the token, but this method is really about applying a transformation